include/RAJA/index/ListSegment.hpp

```diff
@@ -120,8 +120,8 @@ class TypedListSegment
                    Index_type length,
                    camp::resources::Resource resource,
                    IndexOwnership owned = Owned)
-    : m_resource(resource)
   {
+    m_resource = new camp::resources::Resource(resource);
     initIndexData(values, length, owned);
```

Comment:
initIndexData should probably handle allocating m_resource, as it should not be done if Unowned.